blog-content/how-to-instrument-your-react-native-app-with-opentelemetry.md

```diff
@@ -0,0 +1,436 @@
+readingTime: 18
+authorFirstName: Spencer
+authorLastName: Amarantides
+authorTitle: Engineer @ Highlight
+authorTwitter: ''
+authorLinkedIn: 'https://www.linkedin.com/in/spencer-amarantides/'
+authorGithub: 'https://github.com/SpennyNDaJets'
+authorPFP: 'https://avatars.githubusercontent.com/u/17744174?v=4'
+authorWebsite: ''
+tags: Developer Tooling, React Native, Mobile, OpenTelemetry
+metaTitle: "How to instrument your React Native app with OpenTelemetry"
+---
+
+```hint
+Highlight.io is an [open source](https://github.com/highlight/highlight) monitoring platform. If you’re interested in learning more, get started at [highlight.io](https://highlight.io).
+```
+
+OpenTelemetry is an open-source observability framework that provides tools, APIs, and SDKs to collect, process, and export telemetry data like traces, metrics, and logs from applications. It is designed to help developers monitor and troubleshoot distributed systems by providing standardized data formats and integration points for observability tools. If you're new to OpenTelemetry, you can learn more about it [here](https://www.youtube.com/watch?v=ASgosEzG4Pw). 
+
+
+Today, we'll go through a guide to using OpenTelemetry in React Native, including the high-level concepts as well as how to send traces, errors, and logs to your OpenTelemetry backend of choice.
+
+### **Provider**
+
+A provider is the API entry point that holds the configuration for telemetry data. The provider is responsible for setting up the environment and ensuring that all necessary configurations are in place. This can include configuring a vendor specific api key, or something as simple as setting the service name and environment. In our case, we will be using a `TraceProvider` to send traces that will be processed by the Highlight backend and converted to logs, errors, and traces.
+
+In our example, the `TracerProvider` creates a resource, that builds attributes that we want to include with every trace. This includes the `highlight.project_id` to let Highlight know which project the traces below to, and other identifying data, such as service name and environment to help with monitoring and debugging.
+
+Here's a quick example of what this looks like in code:
+
+```typescript
+import { BasicTracerProvider } from '@opentelemetry/sdk-trace-base'
+import { Resource } from '@opentelemetry/resources'
+
+const resource = new Resource({
+	'highlight.project_id': '<YOUR_PROJECT_ID>',
+	'service.name': 'reactnativeapp',
+  'environment': 'production',
+})
+
+const tracerProvider = new BasicTracerProvider({ resource })
+```
+
+### **Exporter**
+
+An exporter sends the telemetry data to the backend. This is where you configure the endpoint and any other necessary settings related to the backend you're sending data to. In our example, we built a custom exporter as a workaround to some OpenTelemetry package issues with the React Native's bundler, Metro. A bundler-based solution is also in progress.
```

Comment:
i like the explanation of why we have to build a custom exporter.

maybe add here a link to the otel exporter that we would otherwise use?